OpenRA.Mods.Common/Activities/PickupUnit.cs

```diff
@@ -65,7 +65,7 @@ protected override void OnFirstRun(Actor self)
 
 		public override bool Tick(Actor self)
 		{
-			if (cargo != carryall.Carryable)
+			if (carryall.IsTraitDisabled || cargo != carryall.Carryable)
```

Comment:
Move this check lower next to carryable.IsTraitDisabled, otherwise this will crash